larq_compute_engine/tflite/tests/bconv2d_test.cc

```diff
@@ -569,8 +714,82 @@ TEST_P(BConv2DOpTest, SimpleTest) {
+            registration, input_shape, packed_filter_shape, input_depth,
+            stride_width, stride_height, bconv_padding, pad_values,
+            dilation_width_factor, dilation_height_factor, activation,
+            num_threads, input_data, packed_filters_data,
+            post_activation_multiplier_data, post_activation_bias_data,
+            m_builtin.GetOutputShape(), builtin_output);
+      }
+    } else {  // Not reading bitpacked input.
+      if (is_reference_registration) {
+        test_lce_op<T, std::int32_t>(
+            registration, input_shape, packed_filter_shape, input_depth,
+            stride_width, stride_height, bconv_padding, pad_values,
+            dilation_width_factor, dilation_height_factor, activation,
+            num_threads, input_data, packed_filters_data,
+            post_activation_multiplier_data, post_activation_bias_data,
+            m_builtin.GetOutputShape(), builtin_output);
+      } else {  // Not the reference op.
+        test_lce_op<T, std::int8_t>(
+            registration, input_shape, packed_filter_shape, input_depth,
+            stride_width, stride_height, bconv_padding, pad_values,
+            dilation_width_factor, dilation_height_factor, activation,
+            num_threads, input_data, packed_filters_data,
+            post_activation_multiplier_data, post_activation_bias_data,
+            m_builtin.GetOutputShape(), builtin_output);
+      }
+    }
+  } else {  // Not writing bitpacked output.
+    if (read_bitpacked_input) {
+      if (is_reference_registration) {
+        test_lce_op<std::int32_t, T>(
+            registration, input_shape, packed_filter_shape, input_depth,
+            stride_width, stride_height, bconv_padding, pad_values,
+            dilation_width_factor, dilation_height_factor, activation,
+            num_threads, input_data, packed_filters_data,
+            post_activation_multiplier_data, post_activation_bias_data,
+            m_builtin.GetOutputShape(), builtin_output);
+      } else {  // Not the reference op.
+        test_lce_op<std::int8_t, T>(
+            registration, input_shape, packed_filter_shape, input_depth,
+            stride_width, stride_height, bconv_padding, pad_values,
+            dilation_width_factor, dilation_height_factor, activation,
+            num_threads, input_data, packed_filters_data,
+            post_activation_multiplier_data, post_activation_bias_data,
+            m_builtin.GetOutputShape(), builtin_output);
+      }
+    } else {  // Not reading bitpacked input.
+      test_lce_op<T, T>(
+          registration, input_shape, packed_filter_shape, input_depth,
+          stride_width, stride_height, bconv_padding, pad_values,
+          dilation_width_factor, dilation_height_factor, activation,
+          num_threads, input_data, packed_filters_data,
+          post_activation_multiplier_data, post_activation_bias_data,
+          m_builtin.GetOutputShape(), builtin_output);
```

Comment:
This sequence of seven calls to `test_lce_op`, varying only in type parameters, is not ideal but I don't see a much better option.